onboardingtaskapp.client/src/components/Customer/List.jsx

```diff
@@ -53,18 +92,10 @@ export default function List() {
                     </tr>
                 </thead>
                 <tbody>
-                    {customers.map(customer => (
-                        <tr key={customer.id}>
-                            <td data-label="Name">{customer.name}</td>
-                            <td data-label="Address">{customer.address}</td>
-                            <td>
-                                <CustomerEdit item={{ customer}} />                               
-                            </td>
-                            <td>
-                                <CustomerDelete item={{ id: customer.id, title: 'Delete Customer', buttonText: 'DELETE', url: 'Customer',  redirect: '' }} />
-                            </td>
-                        </tr>
-                    ))}
+                    {customers.map(customer => 
+                        renderTableRows(customer, handleUpdateSuccess, handleDeleteSuccess)
+                       
+                    )}
```

Comment:
Technically, this is working, however, this implementation is risky because if you will check it specially when you activate inspect element in the browser, you will notice that this page is calling every single item of the data. So for example, if we have 1000 users in here, your application will query 1000 times as well which is not good in terms of consuming APIs and optimisation.

Instead of doing it like this, you can simply re-render the **customers** state by calling a function, for example **renderCustomers()** and from here you can display the customer data using your _mapping_ and then you don't need to pass parameters such as customer, handleUpdateSuccess, and handleDeleteSuccess

Please implement this to all other screen, Thanks
<img width="968" alt="Screenshot 2024-06-18 at 7 35 01 PM" src="https://github.com/harshanajayarathna/OnboardingTaskApp/assets/168311075/6998214d-7b90-4c35-a8c8-817357877843">